Sources/MapboxMaps/Foundation/Camera/CameraAnimator.swift

```diff
@@ -0,0 +1,88 @@
+import UIKit
+
+// MARK: CameraAnimator Class
+public class CameraAnimator: NSObject {
+
+    // MARK: Stored Properties
+
+    /// Instance of the property animator that will run animations.
+    private let propertyAnimator: UIViewPropertyAnimator
+
+    /// Delegate that conforms to `CameraAnimatorDelegate`.
+    private weak var delegate: CameraAnimatorDelegate?
+
+    /// The ID of the owner of this `CameraAnimator`.
+    internal var owner: AnimationOwnerProtocol
+
+    // MARK: Computed Properties
+
+    /// The state from of the animator.
+    public var state: UIViewAnimatingState { return propertyAnimator.state }
+
+    /// Boolean that represents if the animation is running or not.
+    public var isRunning: Bool { return propertyAnimator.isRunning }
+
+    /// Boolean that represents if the animation is running normally or in reverse.
+    public var isReversed: Bool { return propertyAnimator.isReversed }
+
+    /// Value that represents what percentage of the animation has been completed.
+    public var fractionComplete: CGFloat {
+        get { return propertyAnimator.fractionComplete }
+        set { propertyAnimator.fractionComplete = newValue }
+    }
+
+    // MARK: Initializer
+    internal init(delegate: CameraAnimatorDelegate,
+                  propertyAnimator: UIViewPropertyAnimator,
+                  owner: AnimationOwnerProtocol) {
+        self.delegate = delegate
+        self.propertyAnimator = propertyAnimator
+        self.owner = owner
+    }
+
+    // MARK: Functions
+
+    /// Starts the animation
+    public func startAnimation() {
+        propertyAnimator.startAnimation()
+    }
+
+    /// Starts the animation after a `delay` which is of type `TimeInterval`.
+    public func startAnimation(afterDelay delay: TimeInterval) {
+        propertyAnimator.startAnimation(afterDelay: delay)
+    }
+
+    /// Pauses the animation
+    public func pauseAnimation() {
+        propertyAnimator.pauseAnimation()
+    }
+
+    /// Stops the animation and notifies the delegate that the animation finished.
```

Comment:
`notifies the delegate that the animation finished` : This isn't relevant for an end developer.